app/models/form526_submission.rb

```diff
@@ -630,27 +616,33 @@ def cleanup
     EVSS::DisabilityCompensationForm::SubmitForm526Cleanup.perform_async(id)
   end
 
-  def find_user_account_with_icn(records, record_type)
-    records.pluck(:user_account_id).uniq.each do |user_account_id|
-      user_account = UserAccount.find(user_account_id)
-      next if user_account&.icn.blank?
-
-      Rails.logger.info("ICN not found on submission #{id}, " \
-                        "using ICN for user account #{user_account_id} instead (based on #{record_type})")
-      return user_account
-    end
-  end
-
   def get_past_submissions
     Form526Submission.where(user_uuid:).where.not(user_account_id:)
   end
 
-  def get_user_verifications
-    UserVerification.where(idme_uuid: user_uuid)
-                    .or(UserVerification.where(backing_idme_uuid: user_uuid))
-                    .or(UserVerification.where(logingov_uuid: user_uuid))
-                    .or(UserVerification.where(mhv_uuid: user_uuid))
-                    .or(UserVerification.where(dslogon_uuid: user_uuid))
-                    .where.not(user_account_id:)
+  def query_mpi_account
+    mpi_service = MPI::Service.new
+    edipi_response = mpi_service.find_profile_by_edipi(edipi: auth_headers['va_eauth_dodedipnid'])
```

Comment:
we should check for the presence of these different auth headers before querying by them, something like:

> if auth_headers['va_eauth_dodedipnid']
>   query by edipi
> else if auth_headers[name/b-date/ssn] and no result from query by edipi
>   query by attributes
> end
> unless icn.present log 'no icn for user with user_uuid'

Reply:
Also they SHOULD always be there. The user is not allowed to enter the form without them populated. Better to check though just to be safe for sure.